cli/sub_command.go

```diff
@@ -240,8 +261,19 @@ func (c *subCmd) subscribe(p *fisk.ParseContext) error {
 			opts = append(opts, nats.HeadersOnly())
 		}
 
+		// Check if the durable exists and ignore all the other options.
+		var bindDurable bool
 		if len(c.durable) > 0 {
-			opts = append(opts, nats.Durable(c.durable))
+			con, err := js.ConsumerInfo(c.stream, c.durable)
+			if err == nil {
+				bindDurable = true
+				c.jsAck = con.Config.AckPolicy != nats.AckNonePolicy
+				log.Printf("Subscribing to JetStream Stream %q using existing durable %q", c.stream, c.durable)
```

Comment:
I think we can simplify this a bit:

rather than this `bindDurable` and the later if around `bindDurable` we can here set `c.subject=""` and just append the `nats.Bind()` call to `opts`.

Rename `bindDurable` to `shouldAck` and later `c.jsAck=shouldAck`.

no extra logic then and just one call to `js.Subscribe()`

Reply:
When I tried that before, the additional options seem to cause issues even if `nats.Bind(...)` is set. e.g.

```
$ nats sub --stream bar --durable foo
16:27:37 Subscribing to JetStream Stream "bar" using existing durable "push"
nats: error: configuration requests heartbeat to be 5s, but consumer's value is 1s
```

Reply:
ah thats unfortunate :( really dislike this api...leave as is then thanks